### PR TITLE
Add participant, conversation, and turn ID

### LINKED ISSUE #214: Add participant, conversation, and turn ID
- Utterance should get _utterance_id_ as an argument and dialogue should get _conversation_id_ as an argument. _Turn_id_ can correspond to the current number of utterances in dialogue.
- Whatever creates the utterance is responsible for creating the _utterance_id_. Similarly, _conversation_id_ should be created when creating dialogue. 
- This issue includes creating and maintaining participant, conversation, and turn IDs. Sample agents should be extended with these.
- _utterance_id_ could be {user_id}\_{conversation_id}\_{turn_id} as proposed in [this issue](https://github.com/iai-group/DialogueKit/issues/207).

### PR DESCRIPTION
**Pull request by WerLaj**
Part of #214

### CODE REVIEW

**Review by NoB0 — APPROVED** [review]
LGTM.

**Comment by WerLaj** [comment]
Tests will be added in a separate [PR](https://github.com/iai-group/DialogueKit/pull/228)

File: dialoguekit/utils/dialogue_reader.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from dialoguekit.core.intent import Intent
 
 _FIELD_UTTERANCE = "utterance"
+_FIELD_UTTERANCE_ID = "utterance ID"
```
Comment (kbalog) [inline dialoguekit/utils/dialogue_reader.py:12]: call it `utterance_id`

File: dialoguekit/nlg/template_from_training_data.py [hunk 2]
```diff
@@ -108,6 +108,7 @@ def extract_utterance_template(  # noqa: C901
                 if satisfaction_classifier:
                     annotated_utterance = AnnotatedUtterance(
                         text=utterance_record.get("utterance").strip(),
+                        utterance_id=utterance_record.get("utterance ID"),
```
Comment (kbalog) [inline dialoguekit/nlg/template_from_training_data.py:111]: use `_FIELD_*` consts and have those defined only in one place

**Review by kbalog — APPROVED** [review]
LGTM with 2 comments to be addressed

**Pull request merged**